Apply some good security practices

bordenit opened the pull request:
1. Don't run as root.
2. Apply securityContext
3. If it works, I can trust using your container more as it's forced to not run as root in more ways than 1.

cmoulliard commented:
Have you tested on your cluster the change you propose (= dockerfile and helm's deploy manifest) ? @bordenit

bordenit commented:
Sorry, yeah I built my own container from your Dockerfile and use these settings in my Kubernetes deployment.

bordenit commented:
I noticed the build failed on auth as well, but it might be good for CI purposes once the auth is addressed to change this tag as well?

Error: Command failed: DOCKER_BUILDKIT=1 docker buildx build -f Dockerfile -t quay.io/snowdrop/cert-manager-webhook-godaddy:pr-40-0da9b20 -t quay.io/snowdrop/cert-manager-webhook-godaddy:**latest** --platform linux/amd64,linux/arm64 --push .

I normally separate my CI job from the publish job. Publish job could after PR is merged:
quay.io/snowdrop/cert-manager-webhook-godaddy:latest

CI build (before PR is merged) maybe a git reference tag instead of latest.

cmoulliard commented:
Some documentation and Helm changes are needed in order to tell the user that, for security reason, we don't run as root anymore the container, that UID/GUID could be pass as ARG to the docker build and should be, of course, overridden using the Helm chart, etc @bordenit

cmoulliard commented:
> Some documentation and Helm changes are needed in order to tell the user that, for security reason, we don't run as root anymore the container, that UID/GUID could be pass as ARG to the docker build and should be, of course, overridden using the Helm chart, etc @bordenit

Can you have a look please ?

bordenit commented:
@cmoulliard I have moved away from GoDaddy since they have locked down their API to people who have 10 or more domains. I switched to Cloudfare and don't intend to spend more time on this. So, closing issue.
